gobblin-salesforce/src/main/java/org/apache/gobblin/salesforce/SalesforceSource.java

```diff
@@ -252,23 +462,23 @@ private String getDateString(int year) {
     return Utils.dateToString(calendar.getTime(), SalesforceExtractor.SALESFORCE_TIMESTAMP_FORMAT);
   }
 
-  private Histogram parseHistogram(CommandOutput<?, ?> response) throws DataRecordException {
+  /**
+   * Parse the query results into a {@link Histogram}
+   */
+  private Histogram parseHistogram(JsonArray records) {
     log.info("Parse histogram");
+
     Histogram histogram = new Histogram();
 
-    String output;
-    Iterator<String> itr = (Iterator<String>) response.getResults().values().iterator();
-    if (itr.hasNext()) {
-      output = itr.next();
-    } else {
-      throw new DataRecordException("Failed to get data from salesforce; REST response has no output");
-    }
-    JsonArray records = GSON.fromJson(output, JsonObject.class).getAsJsonArray("records");
     Iterator<JsonElement> elements = records.iterator();
     JsonObject element;
+
     while (elements.hasNext()) {
       element = elements.next().getAsJsonObject();
-      histogram.add(new HistogramGroup(element.get("time").getAsString(), element.get("cnt").getAsInt()));
+      String time = element.get("time").getAsString() + ZERO_TIME_SUFFIX;
```

Comment:
In terms of the function itself, whether to append the `ZERO_TIME_SUFFIX` really depends on the `element.get("time")`. My concern is that the function won't be correct if `time` already has a proper suffix.

The solution would be:
1. Leverage `Utils.toDateTimeFormat`, given the input time format
2. Remove the function and put the logic to `getHistogramByDayBucketing`, for it's only used there.